server/src/main/java/io/crate/expression/reference/sys/shard/ShardRowContext.java

```diff
@@ -349,4 +353,28 @@ public Float recoveryFilesPercent() {
             ? null
             : recoveryState.getIndex().recoveredFilesPercent();
     }
+
+    public Long retentionLeasesPrimaryTerm() {
+        try {
+            return indexShard.getRetentionLeaseStats().leases().primaryTerm();
```

Comment:
`getRetentionLeaseStats` can return `null` is it safe to invoke it like this?

Reply:
Currently the expression implementations in `SystemTable` catch npe's, so yes it is safe. But maybe worth handling here to avoid unnecessary stacktrace generations.

I'll adapt it

Reply:
From the code `getRetentionLeaseStats` will never return `null` (compared to `ShardStats` which can return `null`) or do I miss anything?